### DISCUSSION TITLE
Exercícios

### DISCUSSION THREAD

**Discussion by Didone** [Didone/0722-bootcamp-sql#2] [Announcements]
# Exercícios 

Este tópico é um guia de como esperamos que sejam feitas as entregas dos exercícios do **bootcamp de SQL**

- [Exercícios do dia 1](https://github.com/Didone/0722-bootcamp-sql/milestone/1)
- [Exercícios do dia 2](https://github.com/Didone/0722-bootcamp-sql/milestone/2)
- [Exercícios do dia 3](https://github.com/Didone/0722-bootcamp-sql/milestone/3)

## Fork

Primeiramente faça um [fork](https://help.github.com/articles/fork-a-repo) deste repositório 
[![fork](https://user-images.githubusercontent.com/1771724/180793140-8d8d6c4f-8e6e-464b-864b-b2f029c1fd8d.png)](https://github.com/Didone/0722-bootcamp-sql/fork)

## Ambiente

O [guia de contribuição](https://github.com/Didone/0722-bootcamp-sql/blob/e388e3ea404e6685e1480438347c4432381906c8/.github/CONTRIBUTING.md)  descreve as etapas necessárias para a configuração do seu ambiente de desenvolvimento

## Desenvolvimento
Para a solução das tarefas, você deverá modificar os arquivos [engine.py](https://github.com/Didone/0722-bootcamp-sql/blob/main/csvms/engine.py) e [table.py](https://github.com/Didone/0722-bootcamp-sql/blob/main/csvms/table.py) conforme for necessário (de acordo com a descrição da issue).

> Verifique os "TODO"s existentes nesses arquivos, para ajudá-lo a localizar os locais onde existem tarefas a serem implementadas

## Pull Requests

Envie uma _[PR](https://help.github.com/articles/creating-a-pull-request)_ chamada `bootcamp` linkando cada uma das _[issues](https://github.com/Didone/0722-bootcamp-sql/milestones)_ solucionadas.

### Texto

É muito importante adicionar ao texto da descrição da sua _[pull request](https://help.github.com/articles/creating-a-pull-request)_ a lista de todos os  _[issues](https://github.com/Didone/0722-bootcamp-sql/issues)_ solucionadas (não adicione na lista aquelas que você não implementou).

Para isso basta colocar o número da tarefa após o símbolo `#`. Conforme o exemplo abaixo:

```markdown
Issues resolvidas:
-  #7
-  #8 
-  #9 
-  #10 
- ...

**Comment by NashornT** [comment]
Boa tarde, estou com uma duvida na questão da inclusão, como que eu faço para me referenciar a tabela para que eu possa adicionar ou excluir termos nela ?

**Reply by angelicagiacomelilopes** [reply]
Boa tarde, tudo bem @NashornT ?

Como Table é uma classe, você precisa fazer uma instância dela e armazenar em uma variável, assim você conseguirá manipular os dados da tabela.

Exemplo em python:

```
from csvms.table import Table 
table = "lista_frutas" #nome da tabela
tbl = Table(table) # instancia da classe Table
tbl.append('banana', 'doce') # add novo item
print(tbl) # impressão da inclusão
tbl.save() # persistir na tabela Commit
```

**Reply by NashornT** [reply]
Muito obrigado 😄

**Comment by juniorribeiro89** [comment]
Tem algum tutorial de como fazer este procedimento?
Adicionar ao seu motor de interpretação de SQL ([engine.py](https://github.com/Didone/csvms/blob/main/csvms/engine.py)) a funcionalidade de criação de tabelas.

Ao informar a entrada abaixo, uma nova tabela chamada lista_frutas deverá ser criada no seu SGBD

**Reply by juniorribeiro89** [reply]
Adicionar ao seu motor de interpretação de SQL a funcionalidade de criação de tabelas.

**Reply by Didone** [reply]
Boa tarde @juniorribeiro89. Não precisa se preocupar com os exercícios no momento. 
Conforme comentei ontem, teremos um quarto dia (extra) só para falar sobre isso e somente depois é que iniciaremos a "contar" o prazo ( que serão de 2 a 3 semanas) para o desenvolvimento destas tarefas. OK?

**Reply by juniorribeiro89** [reply]
Haaa, Obrigado meu querido, já estava ficando preocupado, gratidão...
